### PR TITLE
8210205: build fails on AIX in hotspot cpp tests (for example getstacktr001.cpp)

### PR DESCRIPTION
**Pull request by backwaterred**
I am creating this PR in response to a build issue brought to my attention by a member of the Adoptium Group. The issue is blocking their build as noted [here](https://github.com/adoptium/temurin-build/issues/2807).

This PR renames instances of `NUMBER_OF_FRAMES` and `MAXIMUM_NUMBER_OF_FRAMES` to `NUMBER_OF_STACK_FRAMES` and `MAXIMUM_NUMBER_OF_STACK_FRAMES` respectively in several files in the directory `jdk11u/test/hotspot/jtreg/vmTestbase/nsk/jvmti/GetStackTrace`. This change fixes the name collision preventing build.

This fix has been tested on my AIX development machine.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8210205](https://bugs.openjdk.java.net/browse/JDK-8210205): build fails on AIX in hotspot cpp tests (for example getstacktr001.cpp)

### Reviewers
 * [Paul Hohensee](https://openjdk.java.net/census#phh) (@phohensee - **Reviewer**)
 * [Andrew Leonard](https://openjdk.java.net/census#aleonard) (@andrew-m-leonard - no project role)
 * [Christoph Langer](https://openjdk.java.net/census#clanger) (@RealCLanger - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk11u pull/24/head:pull/24` \
`$ git checkout pull/24`

Update a local copy of the PR: \
`$ git checkout pull/24` \
`$ git pull https://git.openjdk.java.net/jdk11u pull/24/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 24`

View PR using the GUI difftool: \
`$ git pr show -t 24`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk11u/pull/24.diff">https://git.openjdk.java.net/jdk11u/pull/24.diff</a>

</details>

### CODE REVIEW

**Comment by dholmes-ora** [comment]
Hi Tyler,

You are actually wanting to backport JDK-8210205 to JDK 11u so you need to follow the backport process.

See

https://wiki.openjdk.java.net/display/JDKUpdates/JDK11u

"Everybody is encouraged to submit fixes for OpenJDK 11 by creating a pull request to the **jdk11u-dev** Git repository."

and:

https://wiki.openjdk.java.net/display/JDKUpdates/How+to+contribute+a+fix

Cheers,
David

**Comment by GoeLin** [comment]
@Tyler, welcome to the OpenJDK project.
@David, thanks for posting the advice!

**Comment by GoeLin** [comment]
@tyler, you need to  enable github actions for the repos you want to commit to. In this case, jdk11u-dev.

**Comment by aixtools** [comment]
Thanks for the PR, wherever it really is.

And you may want to make it easier to find the way to be able to post issues. I really did not feel like posting on an Oracle site would be the way to get something noticed here. Having only PR's possible, and what I also notice of - yet another 'contributor' requirement only servers to make it harder to make an issue known - even with the solution (for someone to convert to a PR).

Bottom line: thanks all for fixing this.

**Comment by andrew-m-leonard** [comment]
@backwaterred have you tried the Option1 documented here: https://wiki.openjdk.java.net/display/JDKUpdates/How+to+contribute+a+fix   ?

**Comment by backwaterred** [comment]
Thanks for your comments @dholmes-ora, @GoeLin, and @andrew-m-leonard.

I wanted to get the ball rolling on this issue since it is currently blocking Adoptium's build. They are eager to have it fixed in time for their next GA mid January. I don't believe the regular merge from 11u-dev into 11u would happen in time.

Given the time crunch, I was hoping to solicit the assistance of someone with more authority in the OpenJDK community to move the process along if possible. If this is not possible, I'm happy to proceed as listed in the [document linked above](https://wiki.openjdk.java.net/display/JDKUpdates/How+to+contribute+a+fix)

**Comment by phohensee** [comment]
From https://wiki.openjdk.java.net/display/JDKUpdates/JDK11u:

"During the later stages of a release cycle, the release enters rampdown. The master jdk11u repository contains the latest version of that release, while the jdk11u-dev repository is used to start work on the next release. If a change needs to be pushed to a release in rampdown, push approval can still be requested using the jdk11u-critical-request label. As the name of this tag suggests, this process is intended for fixes such as major regressions that must make the release. More minor bugs and new features should go in the next release being developed in jdk11u-dev. The maintainers may approve with jdk11u-critical-yes, defer to jdk11u-dev or reject altogether. Outstanding approvals for critical fixes can be monitored here. If, and only if, the fix gets approved with jdk11u-critical-yes, it may be pushed to the jdk11u repository. Approved critical fixes show up in this JBS filter (login required)."

If you consider this to be a critical fix, then a pull request against jdk11u seems to be the correct thing to do.

**Comment by andrew-m-leonard** [comment]
@phohensee @backwaterred I think given AIX will not build for jdk-11.0.14, that is a fairly critical reason. The alternative will be for Adoptium to do a "patch" on top of their mirror for the jdk-11.0.14 release.
Depends if anyone else is interested in building AIX for jdk-11.0.14 ?

**Comment by backwaterred** [comment]
Sounds like this falls under the description of critical request (Thanks for the pointer @phohensee).

\label add jdk11u-critical-request

For now, the change is created as an update and not a backport. If there is desire, I'm happy to follow the backport procedure, though I believe that will require creating a new PR.

**Comment by andrew-m-leonard** [comment]
@backwaterred i've manually added the label : https://bugs.openjdk.java.net/browse/JDK-8210205

**Comment by phohensee** [comment]
This PR doesn't have a "backport" label, so perhaps wasn't created using the normal backport process. I'd use that process to create a new PR against a jdk11u fork (instead of a non-critical development jdk11u-dev fork). Patch looks good, btw.

**Review by phohensee — APPROVED** [review]
Lgtm, in case the maintainers are ok with using this PR.

**Review by andrew-m-leonard — APPROVED** [review]
Change looks good

**Comment by GoeLin** [comment]
@Tyler, you should only type /integrate once you got the jdk11u-critical-yes tag on the bug.

**Comment by RealCLanger** [comment]
Hi,
the original commit in head seems to be: https://github.com/openjdk/jdk/commit/45fb6202188628a2dca64074d572586d1f5fd001

So, to get this correctly recognized as a backport, please rename the title of this PR to "Backport 45fb6202188628a2dca64074d572586d1f5fd001" (@backwaterred)

**Comment by tstuefe** [comment]
Hi Tyler,

please, in the future, follow the backport process. It simplifies our work and is actually the fastest way to get things rolling because it streamlines information for us. In particular, it is important to know:
- whether a patch is a clean backport or whether changes had to be made.
- whether and what you tested. This is typically left at the discretion of the authors, but any test gaps would have to be picked by the maintainers. In your case, it would have been interesting to know which AIX versions you tested the build on.

The formalized "Fix Request" comment in JBS allows us to see all that information at a glance.

Thank you,

Thomas

**Review by RealCLanger — CHANGES_REQUESTED** [review]
Fixing the backport status of PR is a prereq.

**Comment by aixtools** [comment]
> Hi Tyler,
> 
> please, in the future, follow the backport process. It simplifies our work and is actually the fastest way to get things rolling because it streamlines information for us. In particular, it is important to know:
> 
>     * whether a patch is a clean backport or whether changes had to be made.
> 
>     * whether and what you tested. This is typically left at the discretion of the authors, but any test gaps would have to be picked by the maintainers. In your case, it would have been interesting to know which AIX versions you tested the build on.
> 
> 
> The formalized "Fix Request" comment in JBS allows us to see all that information at a glance.
> 
> Thank you,
> 
> Thomas

FYI: It was failing on all of AIX 7.1 TL4, AIX 7.1 TL5 and AIX 7.2 TL4 (from memory).

**Comment by aixtools** [comment]
> Hi, the original commit in head seems to be: [openjdk/jdk@45fb620](https://github.com/openjdk/jdk/commit/45fb6202188628a2dca64074d572586d1f5fd001)
> 
> So, to get this correctly recognized as a backport, please rename the title of this PR to "Backport 45fb6202188628a2dca64074d572586d1f5fd001" (@backwaterred)
> 
> I think this PR should be approved and merged before next Tuesday, which is the last day before rampdown. But it needs to be properly recognized as backport by the bots before allowing the integration.

If the only thing holding this back is the title - could anyone change the title before Tuesday, Vacation times are often difficult to make deadlines.

**Comment by tstuefe** [comment]
> > Hi, the original commit in head seems to be: [openjdk/jdk@45fb620](https://github.com/openjdk/jdk/commit/45fb6202188628a2dca64074d572586d1f5fd001)
> > So, to get this correctly recognized as a backport, please rename the title of this PR to "Backport 45fb6202188628a2dca64074d572586d1f5fd001" (@backwaterred)
> > I think this PR should be approved and merged before next Tuesday, which is the last day before rampdown. But it needs to be properly recognized as backport by the bots before allowing the integration.
> 
> If the only thing holding this back is the title - could anyone change the title before Tuesday, Vacation times are often difficult to make deadlines.

Before leaving for vacation @GoeLin added the Fix Request on the JBS issue himself (normally done by the author) and approved it right away. So the only thing left is for the patch author to integrate. AFAIK nobody else can do that. He just has to type "slash integrate". See Skara Documentation: https://wiki.openjdk.java.net/display/SKARA/Pull+Request+Commands#PullRequestCommands-/integrate resp. https://wiki.openjdk.java.net/display/JDKUpdates/How+to+contribute+a+fix

Cheers, Thomas

**Comment by aixtools** [comment]
@tstuefe 

Many thanks for the update. Fingers crossed!

**Comment by backwaterred** [comment]
> Hi Tyler,
> 
> please, in the future, follow the backport process. It simplifies our work and is actually the fastest way to get things rolling because it streamlines information for us. In particular, it is important to know:
> 
>     * whether a patch is a clean backport or whether changes had to be made.
> 
>     * whether and what you tested. This is typically left at the discretion of the authors, but any test gaps would have to be picked by the maintainers. In your case, it would have been interesting to know which AIX versions you tested the build on.
> 
> 
> The formalized "Fix Request" comment in JBS allows us to see all that information at a glance.
> 
> Thank you,
> 
> Thomas

Hi @tstuefe,

I agree with your comment about using the correct process. I recognize that it is helpful, and the best way to get things happening quickly. Thanks for accommodating this atypical request. (Side note: testing was done on AIX 7.1)

A big thanks to everyone for helping this move along quickly.

Happy Holidays all :christmas_tree:

**Review by RealCLanger — APPROVED** [review]

**Comment by aixtools** [comment]
Thanks All!

**Pull request closed without merging**
